### PR TITLE
[OSPRH-6229] Ceilometer gets 403 Forbidden when accessing Swift API

### PR DESCRIPTION
**Pull request by jlarriba**
Depends-On: https://github.com/openstack-k8s-operators/keystone-operator/pull/402
Depends-On: https://github.com/openstack-k8s-operators/keystone-operator/pull/403

### CODE REVIEW

**Comment by jlarriba** [comment]
/hold

I just want to see it running on the CI

**Comment by vyzigold** [comment]
recheck

**Comment by jlarriba** [comment]
Dont recheck please, this is work in progress and even if it passes, it will have wrong logs in the ceilometer pod. In am working on solving it, but needs changes even in keystone.

**Comment by vyzigold** [comment]
I wanted to see if the CI issue Emma detected yesterday is fixed now. This seemed like a good candidate for a recheck. Sorry.

**Comment by jlarriba** [comment]
Ah, then its perfectly ok

**Comment by jlarriba** [comment]
recheck

**Comment by jlarriba** [comment]
recheck

**Comment by jlarriba** [comment]
I have to wait until the changes in keystone are updated in the openstack-operator

**Comment by jlarriba** [comment]
/unhold
/retest
recheck

**Comment by jlarriba** [comment]
recheck

**Comment by jlarriba** [comment]
recheck

**Comment by jlarriba** [comment]
recheck

**Comment by jlarriba** [comment]
recheck

**Review by paramite — APPROVED** [review]

**Comment by jlarriba** [comment]
recheck

File: controllers/ceilometer_controller.go [hunk 1]
```diff
@@ -947,14 +960,48 @@ func (r *CeilometerReconciler) ensureSwiftRole(
...
+	if err != nil {
+		return err
+	}
+
 	err = os.AssignUserRole(
 		log,
 		"SwiftSystemReader",
 		user.ID,
-		"service")
+		project.ID)
```
Comment (csibbitt) [inline controllers/ceilometer_controller.go:973]: :+1:

**Review by csibbitt — APPROVED** [review]

**Comment by jlarriba** [comment]
recheck

**Pull request merged**